packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/interact/FunctionField.tsx

```diff
@@ -56,6 +56,13 @@ export function FunctionField({ worldAbi, functionAbi }: Props) {
   const [txHash, setTxHash] = useState<Hex>();
   const txUrl = blockExplorerTransactionUrl({ hash: txHash, chainId });
 
+  // Tuples are mapped to e.g. "[int16, int16, int16]" to inform the user what to input
+  const inputLabels = functionAbi.inputs.map((input) =>
+    input.type === "tuple"
+      ? `[${(input as { components: readonly AbiParameter[] }).components.map((c) => c.type).join(", ")}]`
```

Comment:
I am still confused why casting is necessary at all, as this type should be narrowed by the `input.type === "tuple"` check

Reply:
I was too. Probably because of union & intersection with `components` not in the base type? Either way, this is fixable by checking for `components` as well, no cast needed then.